### ISSUE TITLE
Unable to download podcasts from FreshRSS feeds

### ISSUE THREAD

**Issue by aab3r** [dvehrs/podget#71]
Trying to download podcasts from FreshRSS feeds:
- In "Authentication", I click on "Allow anonymous reading of the default user’s articles"
- Right-click and save RSS feed.
- Set Podget's serverlist with that feed.
- Result: download error - "No enclosures in feed"
- Feed format: https://freshrss_server/i/?a=rss&get=f_17&rid=651fde29bb414&hours=168
- Opening the feed in Chrome shows that it is correct (all the .mp3 files are there); Tt-rss also gets the entries fine.
- I've tried with "Filename_Formatfix" 4 and 9, with the same results.
- I've tried appending "OPT_CONTENT_DISPOSITION" to the feed, as this has worked before with other feeds, with the same results.

Please, let me know if there's any test I can do. Thanks!

**Comment by dvehrs** [comment]
OK, I understand that we don't always want to share the feeds we follow on tech support logs where they will exist forever but there is literally nothing I can do with the information you have given me because "freshrss_server" is a type of server and not an actual source I can test.  

Now if I were able to test this, the first thing I would do is discard everything below "No enclosures in feed".  

Then I would skip back up the the URL you saved for the feed in Podget's serverlist.  I would download that by hand and see what it is telling me.  Most likely something like:
     wget -O - URL | less  

Now this will require reading through the HTML to find out if it is giving us any hints to why it is having a problem.    Most servers will give some message to why it isn't working when there is a problem.  That is what we are looking for.   Then get a new copy of the URL from the server and see if the results you get from wget are the same as you got from the feed URL from the serverlist.  We want to make sure that it wasn't just a copy & paste error.  I have seen a few of those that can happen depending on which editor you choose to use.  

If the new feed URL you download produces a different result, you can run a slightly different test to see if there is content there for Podget to work with like so:
      wget -O - URL | grep enclosures

If you are still getting a feed without enclosures then we need to consider other potential issues.

Now this idea is a long shot and something I have not seen in quite a while.  At one point, feeds that were hosted on Windows servers would download formatted in UTF16 instead of UTF8.  Podget does have the ability to convert those to the format it understands but it requires you to put the feed URL into a file named serverlist.utf16.  This new serverlist is formatted exactly the same as the default one but just triggers the conversion early on.  

That's probably enough to start testing.  Let me know if it works or if we need to dive deeper.
Good Luck!
Dave

**Comment by aab3r** [comment]
@dvehrs  Thanks for the quick reply!

Sorry, I can't share the freshrss public address, as it is a test server, and not DNS discoverable (yet).
Just in case, I've tried with a FreshRSS demo instance, [https://demo.freshrss.org/i/](https://demo.freshrss.org/i/) with the same results.

 wget -O - URL | less gives:

```
<?xml version="1.0" encoding="UTF-8" ?><rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:media="http://search.yahoo.com/mrss/"
	>
	<channel>
		<title>Salida Principal | FreshRSS</title>
		<link>http://freshrss_server</link>
		<description>Fuente RSS de Salida Principal | FreshRSS</description>
		<pubDate>Sat, 07 Oct 2023 10:27:54 +0000</pubDate>
		<lastBuildDate>Sat, 07 Oct 2023 10:27:54 GMT</lastBuildDate>
		<atom:link href="http://freshrss_server/i/?a=rss" rel="self" type="application/rss+xml" />
		<item>
			<title>Infiltrados</title>
			<link>http://www.ivoox.com/infiltrados-audios-mp3_rf_117310058_1.html</link>
						<dc:creator>Rumore Radio Topo</dc:creator>
			<media:content url="https://www.ivoox.com/infiltrados_mh_117310058_feed_1.mp3" type="audio/mpeg" length="56921922"></media:content>
			<description><![CDATA[Hablamos sobre infiltrados policiales en los movimientos sociales, destapados durante este año, con Adolfo como invitado. Del 40 aniversario de la creación del colectivo Lisístrata, de derechos sexuales y reproductivos, y más.]]></description>
			<pubDate>Fri, 06 Oct 2023 20:18:00 +0000</pubDate>
			<guid isPermaLink="false">1696630387012896</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep30</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep30-audios-mp3_rf_107492941_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep30_mh_107492941_feed_1.mp3" type="audio/mpeg" length="57486150"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Wed, 03 May 2023 10:40:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278867</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep29</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep29-audios-mp3_rf_106854123_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep29_mh_106854123_feed_1.mp3" type="audio/mpeg" length="57174770"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 24 Apr 2023 08:07:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278866</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep28</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep28-audios-mp3_rf_105651469_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep28_mh_105651469_feed_1.mp3" type="audio/mpeg" length="55833539"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 03 Apr 2023 07:50:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278865</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep27</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep27-audios-mp3_rf_105231159_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep27_mh_105231159_feed_1.mp3" type="audio/mpeg" length="55833539"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 27 Mar 2023 11:56:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278864</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep26</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep26-audios-mp3_rf_104906076_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep26_mh_104906076_feed_1.mp3" type="audio/mpeg" length="55701046"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 21 Mar 2023 10:27:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278863</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep24Trovador Urbano T03-Ep25</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep24trovador-urbano-t03-ep25-audios-mp3_rf_104473045_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep24trovador-urbano-t03-ep25_mh_104473045_feed_1.mp3" type="audio/mpeg" length="57466506"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 13 Mar 2023 08:17:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278862</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep24</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep24-audios-mp3_rf_104315923_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep24_mh_104315923_feed_1.mp3" type="audio/mpeg" length="57493673"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Thu, 09 Mar 2023 07:17:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278861</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep22</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep22-audios-mp3_rf_103404118_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep22_mf_103404118_feed_1.mp3" type="audio/mpeg" length="55144451"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 21 Feb 2023 10:23:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278860</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep21</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep21-audios-mp3_rf_102981520_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep21_mf_102981520_feed_1.mp3" type="audio/mpeg" length="54230870"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 13 Feb 2023 11:20:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278859</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep20</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep20-audios-mp3_rf_102607546_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep20_mf_102607546_feed_1.mp3" type="audio/mpeg" length="57010512"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 9 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 06 Feb 2023 08:22:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278858</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep19</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep19-audios-mp3_rf_102242744_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep19_mf_102242744_feed_1.mp3" type="audio/mpeg" length="54529924"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 30 Jan 2023 07:53:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278857</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep18</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep18-audios-mp3_rf_101932408_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep18_mf_101932408_feed_1.mp3" type="audio/mpeg" length="55507531"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 24 Jan 2023 08:32:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278856</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep17</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep17-audios-mp3_rf_101564173_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep17_mf_101564173_feed_1.mp3" type="audio/mpeg" length="53937258"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 17 Jan 2023 07:46:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278855</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep16</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep16-audios-mp3_rf_101159242_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep16_mf_101159242_feed_1.mp3" type="audio/mpeg" length="54548315"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 09 Jan 2023 13:02:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278854</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep15</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep15-audios-mp3_rf_100110567_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep15_mf_100110567_feed_1.mp3" type="audio/mpeg" length="57530453"></media:content>
			<description><![CDATA[Un repaso al mejor rock de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 20 Dec 2022 07:49:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278853</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep14</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep14-audios-mp3_rf_99540692_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep14_mf_99540692_feed_1.mp3" type="audio/mpeg" length="54723984"></media:content>
			<description><![CDATA[Un repaso al mejor rock de todos los tiempos de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 13 Dec 2022 09:35:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278852</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep13</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep13-audios-mp3_rf_97819770_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep13_mf_97819770_feed_1.mp3" type="audio/mpeg" length="56878504"></media:content>
			<description><![CDATA[Un repaso al mejor rock de todos los tiempos de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Mon, 05 Dec 2022 08:01:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278851</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep12</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep12-audios-mp3_rf_97261807_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep12_mf_97261807_feed_1.mp3" type="audio/mpeg" length="55972719"></media:content>
			<description><![CDATA[Un repaso al mejor rock de todos los tiempos de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Wed, 30 Nov 2022 12:15:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278850</guid>
		</item>
		<item>
			<title>Trovador Urbano T03-Ep11</title>
			<link>https://www.ivoox.com/trovador-urbano-t03-ep11-audios-mp3_rf_96709483_1.html</link>
						<dc:creator>Radio Miraflores</dc:creator>
			<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep11_mf_96709483_feed_1.mp3" type="audio/mpeg" length="49882636"></media:content>
			<description><![CDATA[Un repaso al mejor rock de todos los tiempos de la mano de David Calderón, los domingos a las 10 de la noche en Radio Miraflores.]]></description>
			<pubDate>Tue, 22 Nov 2022 08:38:00 +0000</pubDate>
			<guid isPermaLink="false">1696629782278849</guid>
		</item>

	</channel>
</rss>
```

As you can see, no "enclousures" in the log. And something I hadn't noticed: it seems FreshRSS is able to offer a RSS feed, but media files (.mp3 here) are not cached by the server, as, for instance, TT-RSS is able of doing (Podget works fine with those).

So, either I haven't configured FreshRSS correctly to cache media files, or it is not capable of doing it, as the absence of "enclousures" would indicate, right?

Server is a running Debian 11.7, and Podget is running from a Debian 12.1 machine (Podget version 0.9.0-1), so I guess we can discard the encoding issue you mentioned.

**Comment by dvehrs** [comment]
@aab3r 

OK, I think I see the problem.

The problem appears to be created by using FreshRSS which is a feed aggregator to produce feeds that you want to then download with Podget which is a podcast aggregator.  So you're double layering aggregators.  

In this case, the problem appears to be caused by the RSS file that FreshRSS produces is not compatible with the base RSS 2.0 specification for feeds or with the Podcast Namespace project guidelines.

[RSS 2.0 Specification](https://www.rssboard.org/rss-specification)

[Podcast Namespace Extensions](https://github.com/Podcastindex-org/podcast-namespace)  - Look in the Docs subsection.

If we start with the RSS 2.0 specification, we will see that it says that each ITEM tag can have an ENCLOSURE tag for media files.  If we read the Podcast Namespace docs then we see that it adds an ALTERNATE ENCLOSURE tag for when you want to provide more than one media file (translations, etc.).    Podget does not currently support this second tag but I include it here as an example of how RSS feeds could be formatted.

Example of Enclosure and Alternate Enclosure tags from Namespace:
```
<enclosure url="https://example.com/file-0.mp3" length="43200000" type="audio/mpeg" />

<podcast:alternateEnclosure type="audio/mpeg" length="43200000" bitrate="128000" default="true" title="Standard">
    <podcast:source uri="https://example.com/file-0.mp3" />
    <podcast:source uri="ipfs://someRandomMpegFile" />
</podcast:alternateEnclosure>

<podcast:alternateEnclosure type="audio/opus" length="32400000" bitrate="96000" title="High quality">
    <podcast:source uri="https://example.com/file-high.opus" />
    <podcast:source uri="ipfs://someRandomHighBitrateOpusFile" />
</podcast:alternateEnclosure>
```

Now for the sake of completeness, I'm also going to include stuff from the Atom specification which Podget does support.

[RFC 4287 Atom Syndication Format](https://www.rfc-editor.org/rfc/rfc4287)

If you read through it, you will see that  the Atom specification says that media content should be placed in items that begin with the tag formatted like so:
```
<link rel="enclosure" type="audio/mpeg" length="1337" href="http://example.org/audio/ph34r_my_podcast.mp3"/>
```

Now if we step back and look at the RSS feed that FreshRSS produced then we see that it did not follow any of three examples above.  FreshRSS decided to use MEDIA:CONTENT tags.  These tags appear to be part of the attempt by RSS to expand it's support for media files.

[Media RSS Specification](https://www.rssboard.org/media-rss)

What is interesting is the new tags were supposed to supplement the ENCLOSURE tags and not necessarily directly replace them.   Also interesting is that in this case the MEDIA:CONTENT tags do not use any of the expanded capabilities provided by the tag but only have those that are supported by the ENCLOSURE tag.  Also interesting that according to the specification, MEDIA:CONTENT does not have a LENGTH attribute but ENCLOSURE does.  So my guess is this problem is created by the way that FreshRSS is reformatting the content it downloads.  

Example from RSS you downloaded above:
```
<media:content url="https://www.ivoox.com/trovador-urbano-t03-ep11_mf_96709483_feed_1.mp3" type="audio/mpeg" length="49882636"></media:content>
```

I compared that to the results that [FreshRSS Demo](https://demo.freshrss.org/i/) provides.

It had three MEDIA:CONTENT tags and zero ENCLOSURE tags:
```
<media:content url="https://fedoramagazine.org/wp-content/uploads/2023/10/Using_InfluxDB-300x127.jpg" type="image/jpg" length="12718"></media:content>
<media:content url="https://cassidyjames.com/images/blog/guadec-2023/attendees.jpg" type="image/png"></media:content>
<media:content url="https://fedoramagazine.org/wp-content/uploads/2023/03/Test_Days-300x127.jpg" type="image/jpg" length="8537"></media:content>
```

We see the same use of the LENGTH attribute here that we noticed above.

Which leads us to the real question.  Should Podget be extended to support the MEDIA:CONTENT tags?

I'm not sure.  There is potential for problems if a feed includes both the ENCLOSURE tag and the MEDIA:CONTENT tag for the same item.  But will people do that or is it an edge case not worth worrying about?  

I think I may need to fork a test version of Podget to play with to add in support for this tag then run it against the collection of feeds that I use for testing to see what happens.  

It won't happen too quickly but it could be interesting to play with.  I'll keep this open and let you know what I decide or any progress I make.
Dave

**Comment by dvehrs** [comment]
Wait I just notices that the three MEDIA:CONTENT tags in the FreshRSS Demo are for images and not video or audio files.

Should still work for basic testing but we may need to find a feed provided by FreshRSS that does include the file types we would consider significant.   

Also this may add a bit of complexity to the download portion because we will also need to filter on file type.  

This might be a strike against including support for MEDIA:CONTENT tags.

Dave

**Comment by aab3r** [comment]
@dvehrs 

> The problem appears to be created by using FreshRSS which is a feed aggregator to produce feeds that you want to  then download with Podget which is a podcast aggregator. So you're double layering aggregators.

Yes... I know it is a somewhat weird setup, but I need it that way. I have to get about 90 RSS feeds; a wordpress site uses those feeds to create posts (with audio playable widgets) and Podget, from another machine, has to get the real podcasts.
I've been doing it for some time now with TT-RSS, and I'm testing FreshRSS as a replacement.

> It won't happen too quickly but it could be interesting to play with. I'll keep this open and let you know what I decide or any progress I make.

Thanks a lot. And, of course, at your own pace.

> Should still work for basic testing but we may need to find a feed provided by FreshRSS that does include the file types we would consider significant.

I can provide you with a FreshRSS account on my server, as soon as I finish setting it up, if you are interested.

**Comment by dvehrs** [comment]
OK, added initial support for MEDIA:CONTENT tags if you use the OPT_RSS_MEDIACONTENT tag in your serverlist for the feed.  

Seems to work with the FreshRSS Demo feed.

In testing today, I was able to download 7 jpg files,  2 mp3 files, 1 png file and 11 mp3 files.

So not entirely usable but if you were a bit more selective on the feeds you downloaded it could be more usable.

This version has been upload to the DEV branch.  Feel free to test it if you like.

Dave

**Comment by dvehrs** [comment]
OK, had a follow up idea that seems to work.

I've added basic filtering on the MEDIA:CONTENT tags to only grab those for types audio or video.  

This time when I ran a very similar test I downloaded 11 mp4 files, 2 mp3 files and 1 webm file.

So now all files are of the types that Podget cares about.  

I also found a bug in one of the other sed statements that I have corrected.  I will need to do some more extensive testing to verify it is working as it should but it now matches what the documentation comments say it should be.

I've uploaded this version into the DEV branch as well.   This version is probably fairly close to what will become the next release version.  I need to update the new serverlist option in the manual page but other than that it may be down to many rounds of test feeds. 

If you get a chance, give it a test run to see how it works against your server and let me know how it goes. 

Thanks,
Dave

**Comment by aab3r** [comment]
This is... awesome!!

When you said "It won't happen too quickly [...]" I was grateful, as it meant that, sometime in the near future, you'd be able to work on this. I guess the "but it could be interesting to play with" was stronger!! Yayyy!!!

Thanks a lot, Dave. I confirm that the dev branch works fine downloading from FreshRSS.

I've noticed that it only downloads the "general" RSS feed, not the individual channels, and I also have to configure FreshRSS so podcasts are cached. I'll keep on trying to figure this out, but the main part is already done, thanks to you.

**Comment by dvehrs** [comment]
Ok this update is included in the new 0.9.3 version. 
Enjoy!
Dave
